Dequeue coins when exiting

molnard opened the pull request:
fixes: https://github.com/zkSNACKs/WalletWasabi/issues/1059

- [x] Make is stable and responsive
- [x] Check every situation can occur
- [x] Change the text according to the new feature

nopara73 commented:
It doesn't work. I click on the exit and nothing happens.

molnard commented:
Tested it again. It is OK for me. Can you tell me how it is not working?

nopara73 commented:
I clicked on this: ![image](https://user-images.githubusercontent.com/9156103/52152759-da7b7b00-2677-11e9-874b-3bb701588f59.png)

molnard commented:
> I clicked on this: ![image](https://user-images.githubusercontent.com/9156103/52152759-da7b7b00-2677-11e9-874b-3bb701588f59.png)

Fixed it thx! Moved the whole closing out of Global and simplify the process. 

Meanwhile I have seen this error a lot:
```
2019-02-02 12:41:08 INFO AliceClient: Round (400), Alice (26f2585b-7e67-4248-9e4d-15f6f21d84a5): Acquired unsigned CoinJoin: 9ea99e906d3a70d34a0742ba2aeb10336daf394a2e0638f4302a02ff336b2c38.
2019-02-02 12:41:12 ERROR CcjClient: System.NotSupportedException: Coordinator did not add enough value to our outputs in the coinjoin. Missing: 12665 satoshis.
   at WalletWasabi.Services.CcjClient.SignCoinJoin(CcjClientRound ongoingRound, Transaction unsignedCoinJoin) in C:\WORK\WalletWasabi\WalletWasabi\Services\CcjClient.cs:line 359
   at WalletWasabi.Services.CcjClient.TryProcessRoundStateAsync(Int64 ongoingRoundId) in C:\WORK\WalletWasabi\WalletWasabi\Services\CcjClient.cs:line 302
```

In some edge cases Global.DesperateDequeueAllCoinsAsync() is doing something which causing the coin gets banned at the end of the procedure. It is only on Testnet where rounds are going insanely fast but could be a good clue.

nopara73 commented:
- [ ] Cancel doesn't restore the coin queueing state.
- [ ] Not sure what happens when you try to dequeue coins when it's not possible. (>InputReg.) From the GUI, it looks like coins won't get dequeued until the phase completes, which is good, but not even after that (at least from the GUI) but the application stops.  

I think both of these are acceptable. Starting to review the code now.

nopara73 commented:
- [ ] Logs look terrible when mixing and pressing ctrl+c:

![image](https://user-images.githubusercontent.com/9156103/52165003-1c013a00-26fb-11e9-88dd-a1f01881f180.png)

nopara73 commented:
- [ ] Same terrible logs when clicking exit while mixing:

![image](https://user-images.githubusercontent.com/9156103/52165028-67b3e380-26fb-11e9-9777-f304e7800094.png)

nopara73 commented:
Anyway, I let you work on this before I go any further with the review.

molnard commented:
> * [ ]  Cancel doesn't restore the coin queueing state.

Changed the button to Abort. Rollback would require the password Aborting is safer. The user can en-queue later manually.

molnard commented:
> * [ ]  Same terrible logs when clicking exit while mixing:
> 
looks better now.
